### ISSUE TITLE
Problem of "use_checkpoint_attn".

### ISSUE THREAD

**Issue by wlj961012** [JingyunLiang/VRT#9]
I try to reimplement the training part. But I encounter the following problem when I set the param 'use_checkpoint_attn' in self.residual_group2 as True. Could you provide solutions to solve it?
![2022121123637_96838](https://user-images.githubusercontent.com/31840936/153710482-d1f02c6e-e936-4c9f-83a9-7130c3636d69.png)

**Comment by JingyunLiang** [comment]
Bug: PyTorch DistributedDataParallel (DDP) does not support `torch.utils.checkpoint` well. To alleviate the problem, set `find_unused_parameters=False` when `use_checkpoint=True`. If there are other errors, make sure that unused parameters will not change during training loop and set `use_static_graph=True`.

The training code will be put in https://github.com/cszn/KAIR soon.

**Comment by wlj961012** [comment]
Thanks for your reply, I follow your suggestion (set find_unused_parameters=False, and add net._set_static_graph() after DistributedDataParallel (DDP)). However, there is no memory decrease. I am not sure if I understand your suggestion correctly?

**Comment by JingyunLiang** [comment]
You still need to set `use_checkpoint_attn=True` to save memory. Above solution is for the mentioned bug. Use DDP +  find_unused_parameters=False + set_static_graph=True + use_checkpoint_attn=True.

**Comment by wlj961012** [comment]
By Following your suggestion, the memory is decreased. In the supp material, you mentioned that the training batch size is 8, does it means that the batch size of per gpu is 1. However, if I set the batch size of per gpu to 1, the performance is much lower than 31.60 (001_VRT_videosr_bi_REDS_6frames
). Are there extra training strategies, i.e, gradient accumulation?

**Comment by JingyunLiang** [comment]
The batch size is 8.

**Comment by wlj961012** [comment]
Is it the total batch size or batch size per card?

**Comment by JingyunLiang** [comment]
The total batch size is 8 (batch size per card is 1).

**Comment by wlj961012** [comment]
Thanks
